measurement/views.py

```diff
@@ -43,3 +64,243 @@ def get(self, request, *args, **kwargs):
         station_codes = list(stations.values_list("station_code", flat=True))
         context = {"django_context": {"stations_list": {"children": station_codes}}}
         return render(request, "data_report.html", context)
+
+
+class MeasurementDataPagination(PageNumberPagination):
+    """Pagination for measurement data API."""
+
+    page_size = 1000
+    page_size_query_param = "page_size"
+    max_page_size = 10000
+
+
+class MeasurementDataDownloadAPIView(APIView):
+    """API endpoint for downloading measurement data.
+
+    Access to measurement data is controlled via object-level permissions using django-guardian.
+    Users can only access data from Station objects for which they have the `view_measurements`
+    permission assigned. This is configured via django-guardian's object-level permission system,
+    typically by assigning the `view_measurements` permission to users or groups for specific
+    Station instances.
+    """  # noqa E501
+
+    permission_classes = [IsAuthenticated]
+    pagination_class = MeasurementDataPagination
+
+    @property
+    def paginator(self):
+        """Return the paginator instance."""
+        if not hasattr(self, "_paginator"):
+            if self.pagination_class is None:
+                self._paginator = None
+            else:
+                self._paginator = self.pagination_class()
```

Comment:
As `pagination_class` is defined above as not `None`, we can just do:

```suggestion
            self._paginator = self.pagination_class()
```